### DISCUSSION TITLE
Handling render()

### DISCUSSION THREAD

**Discussion by ju5t** [spatie/laravel-livewire-wizard#25] [Ideas]
In a wizard component, I would like to implement `render` myself. We generally implement full page Livewire components with [layout components](https://laravel-livewire.com/docs/2.x/rendering-components#custom-layout). This also saves us from somewhat duplicated controllers; as a controller to only launch the wizard is not what I would do.

I know I can override `render` and implement what we'd like; but it's not my preferred approach. I don't want to copy over the logic from it. What I would like to propose is to move the logic that fetches state into a dedicated method. Or perhaps even a computed Livewire property. It wouldn't change how things work, but it would make implementing your own render method easier.

Another alternative is to make full page steps, but that's a lot of unnecessary dom diffing.

Let me know what you think.

**Comment by freekmurze** [comment]
> I know I can override render and implement what we'd like; but it's not my preferred approach. I don't want to copy over the logic from it. What I would like to propose is to move the logic that fetches state into a dedicated method. Or perhaps even a computed Livewire property. It wouldn't change how things work, but it would make implementing your own render method easier.

Feel free to PR that 👍

**Reply by ju5t** [reply]
Thanks for the quick reply! 🔥
